nats.go

```diff
@@ -609,16 +676,24 @@ type Subscription struct {
 // In case using JetStream, there are multiple ways to ack a Msg:
 //
 //	// Acknowledgement that a message has been processed.
+// Raises the watermark for this consumer
 //	msg.Ack()
 //
+//
 //	// Negatively acknowledges a message.
+// What are the consequences/effects of nacking a message?
 //	msg.Nak()
 //
 //	// Terminate a message so that it is not redelivered further.
+// To just this consumer, or terminates it for all consumers of the stream?
 //	msg.Term()
 //
 //	// Signal the server that the message is being worked on and reset redelivery timer.
 //	msg.InProgress()
+
+//This section on ACKS is useful, but kinda hard to find it under Msg.
+// I have been looking at the docs daily for over a year and never know it was here.
+// Complained multiple times in these notes there's no doc for ACK behavior. I guess some exists after all.
```

Comment:
I guess ack documentation should be on `Ack()`, `Nak()` etc. methods in js.go (and there is, from what I see, although probably can be improved. `Msg` doc could simply link to those methods.